demisto_sdk/commands/doc_reviewer/doc_reviewer.py

```diff
@@ -318,19 +318,23 @@ def remove_punctuation(word):
     def check_word(self, word):
         """Check if a word is legal"""
         # check camel cases
+        word = self.remove_punctuation(word)
+        sub_words = []
+        if '-' in word:
+            sub_words.extend(word.split('-'))
         if not self.no_camel_case and self.is_camel_case(word):
-            word = self.remove_punctuation(word)
-            sub_words = self.camel_case_split(word)
-            for sub_word in sub_words:
-                sub_word = self.remove_punctuation(sub_word)
-                if sub_word.isalpha() and self.spellchecker.unknown([sub_word]):
-                    self.unknown_words[word] = list(self.spellchecker.candidates(sub_word))[:5]
-
+            sub_words.extend(self.camel_case_split(word))
         else:
-            word = self.remove_punctuation(word)
-            if word.isalpha() and self.spellchecker.unknown([word]):
-                self.unknown_words[word] = list(self.spellchecker.candidates(word))[:5]
+            sub_words.append(word)
+
+        self.unknown_words[word] = []
```

Comment:
perhaps a set would be better here?

Reply:
Had to make sure - this'll only change a print, so it's ok to change it to set.